lib/enigma.rb

```diff
@@ -1,8 +1,40 @@
+require 'date'
 class Enigma
+  def rand_key
+    rand(99999)
+  end
 
-  def pad(number)
-    pad_num = number.to_s
+  def zero_pad(key)
+    pad_num = key.to_s
     pad_num.insert(0, "0") until pad_num.size == 5
     pad_num
   end
+
+  def shift_keys(key)
+    keys = []
+    key.chars.each_with_index do |char, index|
+      next if index == 4
+      keys << char + key[index + 1]
+    end
+    keys
+  end
+
+  def date
+    date = Date.today.strftime("%d%m%y")
+  end
+
+  def produce_offset(date)
+    squared_date = date.to_i * date.to_i
+    squared_date.to_s.chars.last(4)
+  end
+
+  def shifts(key, date)
+    key_array = shift_keys(zero_pad(key))
+    offset = produce_offset(date)
+    shifts = []
+    offset.each_with_index do |element, index|
+      shifts << element.to_i + key_array[index].to_i
+    end
+    shifts
+  end
```

Comment:
`shifts` is where it all comes together. First we get our `shift_keys` by giving it the argument of whatever key we have while also `zero_pad`-ing the key, then we get our offset by calling `produce_offset` on the date. 
Once we have these variables we can enumerate using `each_with_index` to add together each of the array's elements based on the index position of each element, and changing each element to be an integer so we can actually do math on them.
Finally we take the results of adding each element and shovel them into an array giving us our four shifts for the cipher!